simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/LocationFragment.kt

```diff
@@ -0,0 +1,263 @@
+
+  private var _binding: FragmentLocationBinding? = null
+  private val binding get() = _binding!!
+
+  private lateinit var navController: NavController
+  private lateinit var nextAction: NavDirections
+  private var isNewYork = false
+
+  private val locationRequestCode = 102
+
+  // Minimum distance between location updates, in meters
+  private val minDistanceUpdates: Float = 10f
+
+  // Minimum time interval between location updates, in milliseconds
+  private val minTimeUpdates = 1000 * 60.toLong()
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    _binding = FragmentLocationBinding.inflate(inflater, container, false)
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    navController = Navigation.findNavController(activity!!, R.id.card_creator_nav_host_fragment)
+    isInNewYorkState()
+
+    binding.checkLocationBtn.setOnClickListener {
+      isInNewYorkState()
+    }
+
+    // Go to next screen
+    binding.nextBtn.setOnClickListener {
+      if (isNewYork) {
+        logger.debug("User navigated to the next screen")
+        nextAction = LocationFragmentDirections.actionNext()
+        navController.navigate(nextAction)
+      } else {
+        val data = Intent()
+        activity!!.setResult(Activity.RESULT_CANCELED, data)
+        activity!!.finish()
+      }
+    }
+
+    // Go to previous screen
+    binding.prevBtn.setOnClickListener {
+      activity!!.onBackPressed()
+    }
+  }
+
+  /**
+   * Show loading screen
+   */
+  private fun showLoading(show: Boolean) {
+    logger.debug("Toggling loading screen")
+    if (show) {
+      binding.loading.visibility = View.VISIBLE
+    } else {
+      binding.loading.visibility = View.GONE
+    }
+  }
+
+  /**
+   * Checks to see if the user has granted required location permissions
+   */
+  private fun getLocation(): Location? {
+    var location: Location? = null
+    logger.debug("Checking for location permission")
+    if (ContextCompat.checkSelfPermission(activity!!, Manifest.permission.ACCESS_COARSE_LOCATION)
+      != PackageManager.PERMISSION_GRANTED
+    ) {
+      logger.debug("Requesting location permission")
+      ActivityCompat.requestPermissions(
+        activity!!,
+        arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION),
+        locationRequestCode
+      )
+    } else {
+      logger.debug("Location permission granted")
+      try {
+        logger.debug("Getting current location")
+        val locationManager = activity!!.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        val isNetworkLocationEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
+
+        if (isNetworkLocationEnabled) {
+          locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, minTimeUpdates, minDistanceUpdates, this)
```

Comment:
You might consider using the method `requestSingleUpdate` here if all you need is a coarse location fix:

https://developer.android.com/reference/android/location/LocationManager#requestSingleUpdate(android.location.Criteria,%20android.location.LocationListener,%20android.os.Looper)

Reply:
Yes, this is better

Reply:
[deprecated](https://developer.android.com/reference/android/location/LocationManager#requestSingleUpdate(java.lang.String,%2520android.app.PendingIntent))

Reply:
Yeah, it's deprecated in Android R, but we can still use it since no one has that yet. They just want us moving to `getCurrentLocation` eventually.